feat: removes agent type-based selection restrictions

av3n93rz opened the pull request:
Simplifies team editing logic by allowing all agent types to be mixed within a team, removing checks and UI feedback that prevented mixing internal and external agents. Improves flexibility in team composition.

## Checklist

- [ ] Follow the [contributor guide](./CONTRIBUTING.md)
- [ ] End-to-end tests pass
- [ ] Unit tests for essential parts of your code, e.g. APIs exposed via SDKs or endpoints
- [ ] Update `./docs`
- [ ] Recognize contributors with "@all-contributors please add <person> for <code, bug, docs, etc>"
- [x] Linked issues [ARKQB-193](https://mckinsey.atlassian.net/browse/ARKQB-193)

david-gombos commented on services/ark-dashboard/ark-dashboard/components/editors/team-editor.tsx:
```diff
@@ -441,19 +377,16 @@ export function TeamEditor({
+                        canAdd={true}
                         isSelected={isSelected}
-                        disabledReason={disabledReason}
+                        disabledReason={''}
```
I think it would make sense to add this as optional parameter instead of an empty string, if there will never be a reason to disable

av3n93rz replied:
Nice catch, I think we can actually remove this and `canAdd`

av3n93rz committed:
feat: removes agent type-based selection restrictions

Simplifies team editing logic by allowing all agent types to be mixed within a team, removing checks and UI feedback that prevented mixing internal and external agents. Improves flexibility in team composition.
